src/adapter/src/catalog.rs

```diff
@@ -2725,6 +2725,13 @@ impl Catalog {
             builtin_table_updates.push(catalog.state.pack_storage_usage_update(&event)?);
         }
 
+        // Now prune rows that we no longer wish to retain
+        catalog
+            .storage()
+            .await
+            .prune_storage_usage(|e| u128::from(e.timestamp()) <= cutoff_ts)
+            .await;
```

Comment:
Nice, this has exactly the right shape! I think we should run this _instead_ of the filter above on L2724. I.e., do the prune, and then unconditionally blat the remaining rows into `builtin_table_updates`. That way if the test passes, we'll know we're actually removing the records from disk.